src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildACppCliProject.cs

```diff
@@ -52,7 +57,27 @@ public void It_builds_and_runs()
                 .HaveStdOutContaining("Hello, World!");
         }
 
-        [FullMSBuildOnlyFact(Skip = "https://github.com/dotnet/sdk/issues/3785")]
+        [FullMSBuildOnlyFact]
+        public void It_builds_and_runs_with_package_reference()
+        {
+            var targetFramework = ToolsetInfo.CurrentTargetFramework + "-windows";
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("NetCoreCsharpAppReferenceCppCliLib")
+                .WithSource()
+                .WithProjectChanges((projectPath, project) => ConfigureProject(projectPath, project, "NewtonSoft.Json","13.0.1", targetFramework, new string[] { "_EnablePackageReferencesInVCProjects" , "IncludeWindowsSDKRefFrameworkReferences" }));
+
+            // build projects separately with BuildProjectReferences=false to simulate VS build behavior
+            new BuildCommand(testAsset, "NETCoreCppCliTest")
+                .Execute("-p:Platform=x64", "-p:EnableManagedpackageReferenceSupport=true", "/bl")
```

Comment:
Nit: Probably don't want to record a binlog except when we're running / debugging a test locally.

```suggestion
                .Execute("-p:Platform=x64", "-p:EnableManagedpackageReferenceSupport=true")
```